test/integration/cx_limit_overload_integration_test.cc

```diff
@@ -0,0 +1,185 @@
+#include "envoy/config/overload/v3/overload.pb.h"
+
+#include "test/integration/integration.h"
+#include "test/test_common/test_runtime.h"
+
+namespace Envoy {
+
+namespace {
+
+envoy::config::overload::v3::OverloadManager overloadManagerProtoConfig(uint32_t max_cx) {
+  const static std::string yaml_config = R"EOF(
+          resource_monitors:
+            - name: "envoy.resource_monitors.global_downstream_max_connections"
+              typed_config:
+                "@type": type.googleapis.com/envoy.extensions.resource_monitors.downstream_connections.v3.DownstreamConnectionsConfig
+                max_active_downstream_connections: {}
+        )EOF";
+  return TestUtility::parseYaml<envoy::config::overload::v3::OverloadManager>(
+      fmt::format(yaml_config, std::to_string(max_cx)));
+}
+
+class GlobalDownstreamCxLimitIntegrationTest : public testing::Test, public BaseIntegrationTest {
+protected:
+  GlobalDownstreamCxLimitIntegrationTest()
+      : BaseIntegrationTest(Network::Address::IpVersion::v4, ConfigHelper::tcpProxyConfig()) {}
+
+  void initializeOverloadManager(uint32_t max_cx) {
+    overload_manager_config_ = overloadManagerProtoConfig(max_cx);
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      *bootstrap.mutable_overload_manager() = this->overload_manager_config_;
+    });
+    initialize();
+  }
+
+  AssertionResult waitForConnections(uint32_t expected_connections) {
+    absl::Notification num_downstream_conns_reached;
+    absl::Mutex lock;
+    test_server_->server().dispatcher().post([this, &lock, &num_downstream_conns_reached,
+                                              expected_connections]() {
+      auto& overload_state = test_server_->server().overloadManager().getThreadLocalOverloadState();
+      const auto& monitor = overload_state.getProactiveResourceMonitorForTest(
+          Server::OverloadProactiveResourceName::GlobalDownstreamMaxConnections);
+      ASSERT_TRUE(monitor.has_value());
+      absl::MutexLock l(&lock);
+      const auto reached_expected_count =
+          [&monitor, expected_connections]() ABSL_EXCLUSIVE_LOCKS_REQUIRED(lock) {
+            return monitor->currentResourceUsage() == expected_connections;
+          };
+      if (timeSystem().waitFor(lock, absl::Condition(&reached_expected_count),
+                               std::chrono::milliseconds(5000))) {
+        num_downstream_conns_reached.Notify();
+      };
+    });
+    return AssertionSuccess();
+  }
+
+private:
+  envoy::config::overload::v3::OverloadManager overload_manager_config_;
+};
+
+TEST_F(GlobalDownstreamCxLimitIntegrationTest, GlobalLimitInOverloadManager) {
+  initializeOverloadManager(6);
+  std::vector<IntegrationTcpClientPtr> tcp_clients;
+  std::vector<FakeRawConnectionPtr> raw_conns;
+  for (int i = 0; i <= 5; ++i) {
+    tcp_clients.emplace_back(makeTcpConnection(lookupPort("listener_0")));
+    raw_conns.emplace_back();
```

Comment:
nit: found it a bit confusing that we were emplacing back empty pointers. Might be more clear if we have the temp ptr that we move to the vector (if we need it).

There are also times when `raw_conns.back()` isn't the most descriptive for a particular connection